### PR TITLE
[air/train] Rename BaseWorkerMixin, only log info torch loop for rank 0

### LINKED ISSUE #20593: [train] rename `BaseWorkerMixin` Actor logs
Currently Ray Train distributed worker methods are executed through a `BaseWorkerMixin` class which may lead to some confusion. Example:
```
(BaseWorkerMixin pid=49676) 2021-11-19 15:40:16,970	INFO torch.py:67 -- Setting up process group for: env:// [rank=0, world_size=2]
(BaseWorkerMixin pid=49682) 2021-11-19 15:40:16,969	INFO torch.py:67 -- Setting up process group for: env:// [rank=1, world_size=2]
```

Ideally we should be able to override these generic worker classes in the logs to something more specific e.g. `DistributedTrainWorker`.

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

This PR

- only prints train_loop info strings (e.g. `train_loop_utils.py:298 -- Moving model to device: cpu`) for rank 0 workers for torch
- renames `BaseWorkerMixin` to `RayTrainWorker` as the name comes up often in output and is more meaningful

## Related issue number

Closes https://github.com/ray-project/ray/issues/20593

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/_internal/worker_group.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 logger = logging.getLogger(__name__)
 
 
-class BaseWorkerMixin:
+class DistributedRayWorker:
```
Comment (matthewdeng) [inline python/ray/train/_internal/worker_group.py:16]: nit: Should we make this specific to Train - `DistributedTrainWorker`? Not sure if it has any benefits of making this generic.

**Review by matthewdeng — APPROVED** [review]

File: python/ray/train/_internal/worker_group.py (hunk 1 above)
Comment (krfricke) [reply python/ray/train/_internal/worker_group.py:16]: @richardliaw what do you think

- DistributedRayWorker
- DistributedTrainWorker
- DistributedRayTrainWorker
- DistributedWorker
- RayTrainWorker
- RayWorker
- TrainWorker

happy either way, let's make it not too long. Slight preference for `RayTrainWorker`, `DistributedRayWorker`, or `DistributedTrainWorker` (in that order)

Comment (richardliaw) [reply python/ray/train/_internal/worker_group.py:16]: RayTrainWorker sounds good to me actually

**Review by richardliaw — APPROVED** [review]
overall looks good;

File: python/ray/train/_internal/worker_group.py (hunk 1 above)
Comment (matthewdeng) [reply python/ray/train/_internal/worker_group.py:16]: Now that you suggest it, it seems like such an obvious choice... 😃

Comment (amogkam) [reply python/ray/train/_internal/worker_group.py:16]: +1 on RayTrainWorker!

**Review by amogkam — APPROVED** [review]
Thanks @krfricke, lgtm!

Though I feel the underlying problem needs to be fixed in Ray's logging...we should not be exposing internal class names to users in the first place.

**Comment by richardliaw** [comment]
> Though I feel the underlying problem needs to be fixed in Ray's logging...we should not be exposing internal class names to users in the first place.

What do you mean here? This is programmable via `__repr__`

**Comment by amogkam** [comment]
@richardliaw I don't think it is. That's why we had to change the class name here

**Comment by richardliaw** [comment]
@amogkam are you sure? There's a test for this - https://github.com/ray-project/ray/blob/395297a9bd0acddb2b39bc2f2bc3c926abbf1769/python/ray/tests/test_output.py#L318-L346

**Comment by amogkam** [comment]
Oh in that case I think we should leverage `__repr__` and also include the rank in the message, rather than changing the class name itself.

Can do this in a follow up PR.

**Comment by krfricke** [comment]
In any case it's good practice to keep the class name and repr name in sync. 

Let's add the rank in a follow-up - that's a great idea and we should actually leverage that in other libraries as well (Ray lightning, XGboost-Ray, LigthGBM-Ray)

**Pull request merged**